server/auth/role.go

```diff
@@ -31,10 +33,24 @@ type Role struct {
 	ConnectionLimit           int32                // rolconnlimit
 	Password                  *ScramSha256Password // rolpassword
 	ValidUntil                *time.Time           // rolvaliduntil
+	id                        RoleID
 }
 
+// RoleID represents a Role's ID. IDs are assigned during load and will be stable throughout the server's current
+// process. IDs are useful for referencing a specific role without using their name, since names can change. This is
+// basically a special OID specific to roles. Eventually, we'll have a proper OID system, but this is a placeholder for
+// now.
+type RoleID uint64
```

Comment:
Should be int32 right?

Reply:
Not in this case, this is replacing the concept of OIDs for users, so it's not restricted to the `uint32` space of an OID. Jason is working on OID caching, and there were some other things we discussed, which this would be compatible with.